### ISSUE TITLE
mongodb: LIKE operator

### ISSUE THREAD

**Issue by ghost** [Impetus/kundera#597]
Hi there,

I'am trying to use the LIKE operator with mongodb. Since the wiki says it works I expected it to work. But I tried it without success. I only get results using the LIKE operator when there is an exact match. 

I tried it with a jpa-query and without lucene.

I couldn't get lucene to work because of missing class. If I try to use lucene I get the following message:
com.impetus.kundera.utils.InvalidConfigurationException: java.lang.ClassNotFoundException: com.impetus.kundera.index.LuceneIndexer

There might be a mistake in IndexingConstants.java because I only found  com.impetus.kundera.index.lucene.Indexer in Kundera core (2.11.1)?

Best Regards

**Comment by chhavigangwal** [comment]
Like operator should work with jpa-query. A sample test case for using like operator with mongoDB 

https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-mongo/src/test/java/com/impetus/client/twitter/TwitterTestBaseMongo.java

Also in order to use lucene you need to add the following to persistence.xml  :

```
<property name="index.home.dir" value="./lucene" />
```

And also add following dependency along with core to use lucene 

```
<dependency>
            <groupId>com.impetus.kundera.core</groupId>
            <artifactId>fallback-impl</artifactId>
            <version>2.11.1</version>
        </dependency>
```

Chhavi

**Comment by ghost** [comment]
But the Twitter example only searches for string parts which are at the beginning of the string (like startsWith() function). This also works for me. 

I didn't find a way to search for any part of a string and not depening on the position of my search string.

**Comment by chhavigangwal** [comment]
Team is working on a fix for this issue. Will soon update.

Chhavi

**Comment by chhavigangwal** [comment]
A fix for the same has been added in current trunk.
Please verify.

Chhavi

**Comment by chhavigangwal** [comment]
Fixed releasing with 2.13
